kubelet/datadog_checks/kubelet/prometheus.py

```diff
@@ -14,52 +14,59 @@
 CONTAINER_LABELS = ['container_name', 'namespace', 'pod_name', 'name', 'image', 'id']
 
 
-class CadvisorPrometheusScraper(PrometheusScraper):
+class CadvisorPrometheusScraperMixin(object):
     """
     This class scrapes metrics for the kubelet "/metrics/cadvisor" prometheus endpoint and submits
     them on behalf of a check.
     """
 
-    def __init__(self, check):
-        super(CadvisorPrometheusScraper, self).__init__(check)
-
-        self.NAMESPACE = 'kubernetes'
-        self.instance_tags = []
-
-        self.ignore_metrics = [
-            'container_cpu_cfs_periods_total',
-            'container_cpu_cfs_throttled_periods_total',
-            'container_cpu_cfs_throttled_seconds_total',
-            'container_cpu_load_average_10s',
-            'container_cpu_system_seconds_total',
-            'container_cpu_user_seconds_total',
-            'container_fs_inodes_free',
-            'container_fs_inodes_total',
-            'container_fs_io_current',
-            'container_fs_io_time_seconds_total',
-            'container_fs_io_time_weighted_seconds_total',
-            'container_fs_read_seconds_total',
-            'container_fs_reads_merged_total',
-            'container_fs_reads_total',
-            'container_fs_sector_reads_total',
-            'container_fs_sector_writes_total',
-            'container_fs_write_seconds_total',
-            'container_fs_writes_merged_total',
-            'container_fs_writes_total',
-            'container_last_seen',
-            'container_start_time_seconds',
-            'container_spec_memory_swap_limit_bytes',
-            'container_scrape_error'
-        ]
-
-        # Filter out system slices to reduce CPU and memory usage of the check
-        self._text_filter_blacklist = ["container_name=\"\""]
+    def __init__(self, *args, **kwargs):
+        super(CadvisorPrometheusScraperMixin, self).__init__(*args, **kwargs)
+
+        # List of strings to filter the input text payload on. If any line contains
+        # one of these strings, it will be filtered out before being parsed.
+        # INTERNAL FEATURE, might be removed in future versions
+        self._text_filter_blacklist = ['container_name=""']
 
         # these are filled by container_<metric-name>_usage_<metric-unit>
         # and container_<metric-name>_limit_<metric-unit> reads it to compute <metric-name>usage_pct
         self.fs_usage_bytes = {}
         self.mem_usage_bytes = {}
 
+    def _create_cadvisor_prometheus_scraper(self, instance):
+        cadvisor_instance = deepcopy(instance)
+        cadvisor_instance.update({
```

Comment:
This technically means that all config options that `PrometheusScraperMixin` understands can now be set for the kubelet check? Should they be documented for the kubelet check or remain "hidden"?

Reply:
Yes, that is now possible. I'll add some documentation to the example config file to let users know they can use all the config options available in the `prometheus` check.